src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -298,13 +298,30 @@ private void RemoveEntry(NodeSearchElement entry)
             var treeStack = new Stack<NodeCategoryViewModel>();
             var nameStack = new Stack<string>(entry.Categories.Reverse());
             var target = libraryRoot;
+            bool isCheckedForClassesCategory = false;
             while (nameStack.Any())
             {
                 var next = nameStack.Pop();
                 var categories = target.SubCategories;
                 var newTarget = categories.FirstOrDefault(c => c.Name == next);
-                if (newTarget == default(NodeCategoryViewModel))
+                if (newTarget == null)
                 {
+                    // Latest item in categories list can be a class.
+                    // All classes are situated in classes container.
+                    // So nameStack should be populated with ClassesDefaultName
+                    // and class name. And checked again.
+                    // For example: "MyAssembly.MyNamespace.ClassCandidate" not found.
+                    // Should be checked for "MyAssembly.MyNamespace.Classes.ClassCandidate"
+                    //
+                    if (!isCheckedForClassesCategory && nameStack.Count == 0)
+                    {
+                        nameStack.Push(next);
+                        nameStack.Push(Configurations.ClassesDefaultName);
+
+                        isCheckedForClassesCategory = true;
+                        continue;
+                    }
+
```

Comment:
I know you have inherited the structure from the main code, maybe we can do this improvement on the `master` branch. The loop `while (nameStack.Any())` can be folded into a `static` method that does the following:

```csharp
private static IEnumerable<NodeCategoryViewModel> GetTreeBranchToNode(
    NodeCategoryViewModel rootNode, NodeSearchElement leafNode)
{
    var nodesOnBranch = new List<NodeCategoryViewModel>();
    var nameStack = new Stack<string>(leafNode.Categories.Reverse());

    while (nameStack.Any())
    {
        // ...
    }
    
    return nodesOnBranch;
}
```

Note that the return result needs to be reversed in the caller, because this method is meant to return nodes on a branch that eventually leads to the given `leafNode` from `rootNode`:

```csharp
var branch = GetTreeBranchToNode(libraryRoot, entry);
var treeStack = new Stack<NodeCategoryViewModel>(branch.Reverse());
```